conformance/lang/expressions/object-constructor-expr/object_methods.balt

```diff
@@ -0,0 +1,332 @@
+Labels: additive-expr, array-type, field-access-expr, int, intersection-type, isolated-object, list-constructor-expr,
+        lock-stmt, member-access-expr, method-call-expr, method-defn, object-constructor-expr, object-field,
+        readonly-type, var
+
+function init() {
+    var obj = isolated object {
+        private int[] val = [1, 2];
+        final readonly & int[] val2 = [3, 4];
+
+        function func1() returns int {
+            int sum = self.val2[1];
+            lock {
+                return sum + self.val[0];
+            }
+        }
+    };
+    io:println(obj.func1()); // @output 5
+}
+
+Test-Case: error
+Description: Test accessing the self variable without using a lock statement when the object-constructor-expr is
+             explicitly isolated and self is part of a field-access-expr of the form self.f, where f is the name of a
+             non-isolated field.
+Labels: array-type, field-access-expr, int, isolated-object, list-constructor-expr, member-access-expr, method-defn,
+        object-constructor-expr, object-field, var
+
+function errorFunction() {
+    var _ = isolated object {
+        private int[] val = [1, 2];
+
+        function func1() returns int {
+            return self.val[0]; // @error invalid access of a mutable field of an isolated object outside a lock statement
+        }
+    };
+}
+
+Test-Case: output
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: additive-expr, array-type, field-access-expr, int, intersection-type, isolated-function, list-constructor-expr,
+        member-access-expr, method-call-expr, method-defn, object-constructor-expr, object-field, readonly-type, var
+
+function init() {
+    var obj = object {
+        final readonly & int[] val = [1, 2];
+
+        isolated function func1() returns int {
+            return self.val[0];
+        }
+
+        isolated function func2() returns int {
+            return self.val[1] + self.func1();
+        }
+    };
+    io:println(obj.func2()); // @output 3
+}
+
+Test-Case: error
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: additive-expr, array-type, field-access-expr, int, intersection-type, isolated-function, list-constructor-expr,
+        member-access-expr, method-call-expr, method-defn, object-constructor-expr, object-field, readonly-type, var
+
+function errorFunction() {
+    var _ = object {
+        final readonly & int[] val = [1, 2];
+
+        function func1() returns int {
+            return self.val[0];
+        }
+
+        isolated function func2() returns int {
+            return self.val[1] + self.func1(); // @error invalid invocation of a non-isolated function in an 'isolated' function
+        }
+    };
+}
+
+Test-Case: output
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
```

Comment:
Shall we differentiate in the descriptions as to how each of the isolated-qualified method defs are tested

Reply:
5 tests with the same description